tests/test_ecs.py

```diff
@@ -88,6 +88,58 @@ def patch_calculate_task_resource_requirements(
     return _calculate_task_resource_requirements(task_definition)
 
 
+def create_log_stream(session, run_task, *args, **kwargs):
+    """
+    When running a task, create the log group and stream if logging is configured on
+    containers.
+
+    See https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_awslogs.html
+    """
+    tasks = run_task(*args, **kwargs)
+    if not tasks:
+        return tasks
+    task = tasks[0]
+
+    ecs_client = session.client("ecs")
+    logs_client = session.client("logs")
+
+    task_definition = ecs_client.describe_task_definition(
+        taskDefinition=task.task_definition_arn
+    )["taskDefinition"]
+
+    for container in task_definition.get("containerDefinitions", []):
+        log_config = container.get("logConfiguration", {})
+        if log_config:
+            if log_config.get("logDriver") != "awslogs":
+                continue
+
+            options = log_config.get("options", {})
+            if not options:
+                raise ValueError("logConfiguration does not include options.")
+
+            group_name = options.get("awslogs-group")
+            if not group_name:
+                raise ValueError(
+                    "logConfiguration.options does not include awslogs-group"
+                )
+
+            if options.get("awslogs-create-group") == "true":
+                logs_client.create_log_group(logGroupName=group_name)
+
+            stream_prefix = options.get("awslogs-stream-prefix")
+            if not stream_prefix:
+                raise ValueError(
+                    "logConfiguration.options does not include awslogs-stream-prefix"
+                )
+
+            logs_client.create_log_stream(
+                logGroupName=group_name,
+                logStreamName=f"{stream_prefix}/{container['name']}/{task.id}",
```

Comment:
is it worth adding a utility function for reuse between the two places?
```py
def make_log_stream_name(stream_prefix, container_name, task_id):
  return f"{stream_prefix}/{container_name}/{task_id}"
```

Or at least adding a comment that these two places need to work the same way?

Reply:
Perhaps, but if it changes in one place and not the other I _want_ the tests to fail. If it's consolidated into a single place, we're more likely to accidentally change it.

Ideally this would just be handled correctly in moto, I think the best move here is to open an issue upstream and consider contributing this behavior there.

Reply:
Oops, didn't realize this was in a test file!

Which assert verifies that logs have been read? This one?
```
assert "Failed to read log events" not in caplog.text
```

Reply:
Unfortunately there's not a clear way to assert this. We _do_ read from the correct log group (or this test would throw an error) but we since moto does not actually read the logs events we write during setup, we're missing an assertion that we stream the log events.